packages/attestation-service/src/requestHandlers/attestation.ts

```diff
@@ -50,36 +51,48 @@ class AttestationRequestHandler {
     this.key = getAttestationKey(this.attestationRequest)
   }
 
-  async validateAttestationRequest() {
+  async findOrValidateRequest(): Promise<AttestationModel | null> {
     const { account, issuer } = this.attestationRequest
 
     const address = getAccountAddress()
     if (!eqAddress(address, issuer)) {
       Counters.attestationRequestsWrongIssuer.inc()
-      throw new Error(`Mismatching issuer, I am ${address}`)
+      throw new ErrorWithResponse(`Mismatching issuer, I am ${address}`, 422)
     }
 
-    const attestationRecord = await findAttestationByKey(this.key, {
+    const attestation = await findAttestationByKey(this.key, {
       logging: this.sequelizeLogger,
     })
 
-    if (attestationRecord && attestationRecord.status !== AttestationStatus.NotSent) {
-      Counters.attestationRequestsAlreadySent.inc()
-      throw new Error(ATTESTATION_ALREADY_SENT_ERROR)
+    // Re-requests for existing attestations skip the on-chain check.
+    if (attestation) {
+      Counters.attestationRequestsRerequest.inc()
+      return attestation
     }
 
-    if (!isDevMode()) {
-      const attestations = await kit.contracts.getAttestations()
-      const state = await attestations.getAttestationState(this.key.identifier, account, issuer)
+    if (isDevMode()) {
+      return attestation
+    }
 
-      if (state.attestationState !== AttestationState.Incomplete) {
-        Counters.attestationRequestsWOIncompleteAttestation.inc()
-        throw new Error(NO_INCOMPLETE_ATTESTATION_FOUND_ERROR)
+    // Check the on-chain status of the attestation. If it's marked Complete, don't do it.
+    // If it's missing, the full node could be behind by a block or two. Try a few times before erroring.
+    const attestations = await kit.contracts.getAttestations()
+    for (let i = 0; i < 4; i++) {
+      const state = await attestations.getAttestationState(this.key.identifier, account, issuer)
+      if (state.attestationState === AttestationState.Incomplete) {
```

Comment:
what is the return value of `state` if the attestation doesn't exist yet? 

i feel like you'd get an error...in which case you need a try/catch to all for retry.

or if it returns null, this `null.attestationState` would cause an exception as well.

Reply:
hmm possibly! Edited to catch this.